spring-data-cassandra/src/test/java/org/springframework/data/cassandra/repository/support/BasicMapIdUnitTests.java

```diff
@@ -41,4 +55,471 @@ public void testMapConstructor() {
 		assertThat(map.get("field1")).isEqualTo(basicMapId.get("field1"));
 		assertThat(map.get("field2")).isEqualTo(basicMapId.get("field2"));
 	}
+
+
+	@Test
+	public void testOne() throws Exception {
+
+		BasicMapId basicMapId = new BasicMapId();
+		basicMapId.putAll(basicMapId);
+
+		assertEquals(0, basicMapId.size());
+		assertTrue(basicMapId.isEmpty());
+	}
+
+
+	@Test
+	public void testTwo() throws Exception {
+
+		BasicMapId basicMapId = new BasicMapId();
+		int intOne = basicMapId.size();
+
+		assertEquals(0, basicMapId.size());
+		assertTrue(basicMapId.isEmpty());
+
+		assertEquals(0, intOne);
+	}
+
+
+	@Test
+	public void testThree() throws Exception {
+
+		BasicMapId basicMapId = (BasicMapId) BasicMapId.id("#xAD]-}", (Serializable) "#xAD]-}");
+
+		assertEquals(1, basicMapId.size());
+		assertFalse(basicMapId.isEmpty());
+
+		Serializable serializable = basicMapId.put("", (Serializable) "#xAD]-}");
+
+		assertEquals(2, basicMapId.size());
+		assertFalse(basicMapId.isEmpty());
+
+		assertNull(serializable);
+
+		String string = (String) basicMapId.put("", (Serializable) "#xAD]-}");
+
+		assertEquals(2, basicMapId.size());
+		assertFalse(basicMapId.isEmpty());
+
+		assertNotNull(string);
+		assertEquals("#xAD]-}", string);
+	}
+
+
+	@Test
+	public void testFour() throws Exception {
+
+		BasicMapId basicMapId = new BasicMapId();
+		HashMap<String, String> hashMap = new HashMap<String, String>();
```

Comment:
As Unit Test methods in my opinion are not meant to get reused or inherited from on a regular basis I didn't suppose this to be a problem or odd.
However, I've changed the line.